### PR TITLE
Add cookbook entry for generating TLS by owning a domain.

### PR DESCRIPTION
**Pull request by andrey-git**
**Description:**

Add cookbook entry for generating TLS by owning a domain.

Usecase:
* Your HA instance is not exposed to the internet.
* You have a public domain (doesn't need to point anywhere)
* Your router supports DNSMasq or otherwise allows to set custom name->IP mappings.

### CODE REVIEW

File: source/_cookbook/tls_domain_certificate.markdown [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+-------------------------------------------------------------------------------
+Press Enter to Continue
+```
+
+* Deploy the value to TXT field using your domain registar.
+
+* Go to a site that queries domain record. For example [this one](https://mxtoolbox.com/TXTLookup.aspx) and look if it sees your brand new TXT field (Don't forget to enter the full domain: `_acme-challenge.mydomain.com`)
+* Press Enter at certbot prompt.
+
+## Make mydomain.com point to your Homeassistant instance
```
Comment (fabaff) [inline source/_cookbook/tls_domain_certificate.markdown]: `Home Assistant`, please

File: source/_cookbook/tls_domain_certificate.markdown [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+If your router uses DNSMasq (for example DDWRT) add the following line to DNSMasq options:
+```
+address=/mydomain.com/<hass ip>
+```
+
+## Edit your Homeassistant configuration to use your certificates
```
Comment (fabaff) [inline source/_cookbook/tls_domain_certificate.markdown]: dito

File: source/_cookbook/tls_domain_certificate.markdown [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+```bash
+$ mkdir certbot
+$ cd certbot
+$ wget https://dl.eff.org/certbot-auto
+$ chmod a+x certbot-auto
+$ sudo ./certbot-auto --manual certonly --preferred-challenges dns -d "mydomain.com" --email your@email.address
+```
+
+* Agree to Tos
+* Choose whether to share your email with EFF
```
Comment (fabaff) [inline source/_cookbook/tls_domain_certificate.markdown]: Would be better to "Electronic Frontier Foundation". Corresponding with the output from `certbot`

File: source/_cookbook/tls_domain_certificate.markdown [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+## Run certbot
+```bash
+$ mkdir certbot
+$ cd certbot
+$ wget https://dl.eff.org/certbot-auto
+$ chmod a+x certbot-auto
+$ sudo ./certbot-auto --manual certonly --preferred-challenges dns -d "mydomain.com" --email your@email.address
+```
+
+* Agree to Tos
```
Comment (fabaff) [inline source/_cookbook/tls_domain_certificate.markdown]: `certbot` uses "Terms of Service" and "Tos".

File: source/_cookbook/tls_domain_certificate.markdown [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+sharing: true
+footer: true
+ha_category: Infrastructure
+---
+
+If your Home Assistant instance is only accessible from your local network you can still protect the communication between your browsers and the frontend with SSL/TLS. You can use [Self-sign certificate](/cookbook/tls_self_signed_certificate/) but your browser will present a warning and some https-only features might not work.
+
+## Prerequirement for this guide:
+* Your Homeassistant instance is not exposed to the intenet. If it is - use [this guide]({{site_root}}/blog/2015/12/13/setup-encryption-using-lets-encrypt/)
+* You control a public domain name. The domain doesn't have to point to a site. A domain controlled by a *trusted* friend will do. (A friend you trust not to MItM you)
```
Comment (fabaff) [inline source/_cookbook/tls_domain_certificate.markdown]: I would suggest to use all capital letters for MITM.

File: source/_cookbook/tls_domain_certificate.markdown [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+$ sudo ./certbot-auto --manual certonly --preferred-challenges dns -d "mydomain.com" --email your@email.address
+```
+
+* Agree to Tos
+* Choose whether to share your email with EFF
+* Agree to your IP being logged
+
+You will get the following text:
+
+```
```
Comment (fabaff) [inline source/_cookbook/tls_domain_certificate.markdown]: You could use `bash` or `text` for format.

File: source/_cookbook/tls_domain_certificate.markdown [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+sidebar: true
+comments: false
+sharing: true
+footer: true
+ha_category: Infrastructure
+---
+
+If your Home Assistant instance is only accessible from your local network you can still protect the communication between your browsers and the frontend with SSL/TLS. You can use [Self-sign certificate](/cookbook/tls_self_signed_certificate/) but your browser will present a warning and some https-only features might not work.
+
+## Prerequirement for this guide:
```
Comment (andrey-git) [inline source/_cookbook/tls_domain_certificate.markdown]: Done

File: source/_cookbook/tls_domain_certificate.markdown [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+* Your Homeassistant instance is not exposed to the intenet. If it is - use [this guide]({{site_root}}/blog/2015/12/13/setup-encryption-using-lets-encrypt/)
```
Comment (andrey-git) [inline source/_cookbook/tls_domain_certificate.markdown]: Done

File: source/_cookbook/tls_domain_certificate.markdown (hunk 5 above)
Comment (andrey-git) [inline source/_cookbook/tls_domain_certificate.markdown]: Done

File: source/_cookbook/tls_domain_certificate.markdown (hunk 4 above)
Comment (andrey-git) [inline source/_cookbook/tls_domain_certificate.markdown]: Done

File: source/_cookbook/tls_domain_certificate.markdown (hunk 3 above)
Comment (andrey-git) [inline source/_cookbook/tls_domain_certificate.markdown]: Done

File: source/_cookbook/tls_domain_certificate.markdown (hunk 6 above)
Comment (andrey-git) [inline source/_cookbook/tls_domain_certificate.markdown]: Done

File: source/_cookbook/tls_domain_certificate.markdown (hunk 1 above)
Comment (andrey-git) [inline source/_cookbook/tls_domain_certificate.markdown]: Done

File: source/_cookbook/tls_domain_certificate.markdown (hunk 2 above)
Comment (andrey-git) [inline source/_cookbook/tls_domain_certificate.markdown]: Done

File: source/_cookbook/tls_domain_certificate.markdown [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+
+## Edit your Homeassistant configuration to use your certificates
+```yaml
+http:
+  api_password: YOUR_SECRET_PASSWORD
+  base_url: https://mydomain.com:8123
+  ssl_certificate: /etc/letsencrypt/live/mydomain.com/fullchain.pem
+  ssl_key: /etc/letsencrypt/live/mydomain.com/privkey.pem
+```
+Make sure the files are accessible by your homeassisatnt user.
```
Comment (andrey-git) [inline source/_cookbook/tls_domain_certificate.markdown]: Done

**Comment by andrey-git** [comment]
Addressed all comments

**Pull request merged**